### DISCUSSION TITLE
Kata 4 Ejercicio 1

### DISCUSSION THREAD

**Discussion by MonserratAlvarezM** [LaunchX-InnovaccionVirtual/CursoIntroPython#87] [Q&A] [answered]
Hola compañaros ! 

Ya encontré el error, aparte de tener mal las definiciones de las variables, no actualice el blucle o.o por eso me seguía dando el mismo error !!! Mil gracias!! 

Tengo un problema con el ejercicio del bucle con for , el error que me sale es este: 

![image](https://user-images.githubusercontent.com/99102171/153503155-d812122f-653f-44f2-be83-097617d1c676.png)

Ya le intenté mil formas , pero no entiendo a que se refiere el error

**Comment by nosoyhector** [comment]
Una pregunta similar, con respuestas que podrían aclarar tus dudas
[https://github.com/LaunchX-InnovaccionVirtual/CursoIntroPython/discussions/76]

**Reply by MonserratAlvarezM** [reply]
Mil gracias, no vi la entrada, vi otras pero del ejercicio 2 ://. Te agrdezco !

**Comment by Leonelney** [comment]
Cuando creas for anidados debes cambiar el nombre de la variable al que se le asignan los valores de la lista, en la linea 2 quieres guardar cada objeto de la lista key_word en la misma variable llamada key_word y eso no es posible.

**Reply by MonserratAlvarezM** [reply]
Es que eso es lo que yo no comprendo, trato de hacer incluso el cambio pero me sale el mismo error :((

**Reply by MonserratAlvarezM** [reply]
Ya encontré el error ... No actualice el blucle o.o !!! Mil gracias!!

**Comment by MarcosSilvaG** [comment]
parece ser un problema con el nombre que asiganas a la lista y al "elemento" que recorrerá dicha lista con el for, 
en tu caso prueba poniendolo de la siguiente manera.
```
for key_word in text:
    for word in (nombre_de_tu_lista):
        if word in key_word:
            pint(key_word)
```

**Reply by MonserratAlvarezM** [reply]
Te lo agradezco! Lo intenté pero me sigue saliendo el error de "str" y es donde no capto como a qué se refiere :((

**Reply by MarcosSilvaG** [reply]
pasa captura de tus listas :D

**Reply by MonserratAlvarezM** [reply]
![image](https://user-images.githubusercontent.com/99102171/153510789-4407261b-35a0-4680-bb34-a3782ed0c026.png)

**Reply by MonserratAlvarezM** [reply]
Ya encontré el error ... No actualice el blucle o.o !!! Mil gracias!!

**Reply by DiegoPenalozaJuarez** [reply]
En tu segunda linea del codigo esta mal tu for, tienes tu lista en parentesis (), no entiendo el porque. Quitale los parentesis para que lo identifique como elemento tu lista.

**Comment by Dfredude** [comment]
- Es incorrecto haber hecho esto:
```python
for key_word in text:
    for key_word in key_word:
```
Al hacer el loop interior, vuelves a definir la variable `key_word` (La modificas) . Empiezas iterandola y despues la redefines, creando un caos, y corrompiendo la estructura del programa. Leccion: Nunca uses el mismo nombre de variable en la estructura simple del bucle `for <variable> in <otra_variable>: `

- En fin para corregir el error 
```
TypeError: str is not callable
```
Pusiste: `print (key_word)`
Deberia de ser: `print(key_word)`
Es un simple espacio de diferencia. El compilador piensa que queres llamar una funcion con el `str` dentro de los parentesis. Solo elimina ese espacio y al menos ese error se quitara. Espero te haya ayudado!

**Reply by MonserratAlvarezM** [reply]
Si :S ya revisé y corregí el primer error, muchas gracias! Pero sigue apareciendo lo mismo incluso quitando el espacio 

![image](https://user-images.githubusercontent.com/99102171/153510770-a057e128-9155-4516-a21f-8d56f5b65ce4.png)

**Reply by MonserratAlvarezM** [reply]
Ya encontré el error ... No actualice el blucle o.o !!! Mil gracias!!

**Comment by doguedogue** [comment]
Aunque me parece que tienes invertida la búsqueda de la frase y la palabra ( **in** )

```
frase = "Me gusta el cappuccino"
palabra = "cappuccino"
if palabra in frase:
    print("Palabra: '" + palabra + "' encontrada en la frase: '"+frase+"'")
else:
    print("Palabra: '" + palabra + "' NO encontrada en la frase: '"+frase+"'")

if frase in palabra:
    print("Frase: '" + frase + "' encontrada en la palabra: '"+palabra+"'")
else:
    print("Frase: '" + frase + "' NO encontrada en palabra: '"+palabra+"'")
```

![image](https://user-images.githubusercontent.com/23409026/153515757-f3959c12-b2e7-4032-8f6f-ef248bedd609.png)

**Comment by AdolfooGc** [comment, marked as answer]
> Hola compañaros !
> 
> Ya encontré el error, aparte de tener mal las definiciones de las variables, no actualice el blucle o.o por eso me seguía dando el mismo error !!! Mil gracias!!
> 
> Tengo un problema con el ejercicio del bucle con for , el error que me sale es este:
> 
> ![image](https://user-images.githubusercontent.com/99102171/153503155-d812122f-653f-44f2-be83-097617d1c676.png)
> 
> Ya le intenté mil formas , pero no entiendo a que se refiere el error


> 

Hola como estas? Oye una pregunta como actualizaste el bucle? es que me aparece el mismo error :/

**Reply by MonserratAlvarezM** [reply]
Arriba en la terminal aparece el boton de "Restart", le das y le dices que sí y listo con eso queda!

**Comment by 3rickDJ** [comment]
@MonserratAlvarezM  nos ayudarías a marcar tu pregunta como respondida :cake: para que se responda más rápido a quienes aún lo necesitan :100: Grax :stadium:  .
